kratos/utilities/read_materials_utility.cpp

```diff
@@ -156,45 +135,58 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
         // We don't just copy the values, we do some tyransformation depending of the destination variable
         if(KratosComponents<Variable<double> >::Has(variable_name)) {
             const Variable<double>& variable = KratosComponents<Variable<double>>().Get(variable_name);
-            p_prop->SetValue(variable, value.GetDouble());
+            if (!value.IsNull())
+                pNewProperty->SetValue(variable, value.GetDouble());
         } else if(KratosComponents<Variable<bool> >::Has(variable_name)) {
             const Variable<bool>& variable = KratosComponents<Variable<bool>>().Get(variable_name);
-            p_prop->SetValue(variable, value.GetBool());
+            if (!value.IsNull())
+                pNewProperty->SetValue(variable, value.GetBool());
         } else if(KratosComponents<Variable<int> >::Has(variable_name)) {
             const Variable<int>& variable = KratosComponents<Variable<int>>().Get(variable_name);
-            p_prop->SetValue(variable, value.GetInt());
+            if (!value.IsNull())
```

Comment:
I am not sure if this should be here ...
It skips setting the value if it is null => just remove it if you don't need it ...?

Either way, I don't see how this is related to the SubProperties. The behavior of this class affects a lot of ppl and should be a different PR

Reply:
I cannot split this PR, many changes and I am managing conflicts every week, that's why the duplicated code. We should discuss this here

Reply:
This is due to a decision that wasn't taken by me, and it is supposed to be a consensus. Please, not only ask me, ask to @KratosMultiphysics/technical-committee too, because sometimes I am only following external requests

Reply:
ok then I remove my blocker, sorry for he noise
still I am not loving this ...

Reply:
same open discussion... null or zero or better option

Reply:
if the zero option is finally accepted, then we will not need to allow the null setting (and this lines may be reverted)

Reply:
I do not follow this part of the discussion. Why is the IsNull needed?

Reply:
Actually i would remove this constructor from this PR and eventually add as a separate PR to be discussed (i consider this a rather big change)

Reply:
but what happens in case of null?

Reply:
If nothing happens in case of null i would definitely agree with @philbucher that a null variable should not be inputed.

just take into account that it is not clear at all who could set it at a later stage. Definitely NOT a Constitutive Law!!

Reply:
Why not? The composite constitutive law is in charge of the mixture.

Reply:
The properties are shared btw elements of a modelpart
the CLaws on the other hand are members of the elements (usually one per GP in StrucutralMechanics)

=> what would be possible (and what I thought you want to do) is to set some values of the Properties from the SubProperties.

Also I agree with @RiccardoRossi that is is not clear from where the values on the Properties are set. Here the json-pointers could work, then it would be clear

Is there a technical reason you need to indicate which values in the properties will be calculated or is this just for clarity?

Reply:
ok, I understand. So we can remove all `null` or `@auto` variables from the properties. They could be defined at the post-processing stage.

Reply:
> The properties are shared btw elements of a modelpart
> the CLaws on the other hand are members of the elements (usually one per GP in StrucutralMechanics)
> 
> => what would be possible (and what I thought you want to do) is to set some values of the Properties from the SubProperties.
> 
> Also I agree with @RiccardoRossi that is is not clear from where the values on the Properties are set. Here the json-pointers could work, then it would be clear
> 
> Is there a technical reason you need to indicate which values in the properties will be calculated or is this just for clarity?

The technical reason was to make visible which variables can be asked  and to provide an answer if someone needs this derived property(ies) for example, the YOUNG modulus of a composite material depends on the Young modulus of its components (defined in the subproperties). Moreover, we can define `young_longitudinal` and `young_transversal` for unidirectional composites.
I think it is a good idea to setting them at the moment of reading the derived property. But the question still remains: which method will be in charge of setting these variables?